crates/fluvio-stream-dispatcher/src/store/memory.rs

```diff
@@ -0,0 +1,534 @@
+    where
+        S: Spec,
+    {
+        let store = self.get_store::<S>().await;
+        let key = value.metadata.name.clone();
+
+        let mut k8_obj: K8Obj<S> = K8Obj::new(key.clone(), value.spec);
+
+        let metadata = value.metadata;
+
+        k8_obj.metadata = ObjectMeta {
+            name: metadata.name,
+            owner_references: metadata.owner_references,
+            labels: metadata.labels,
+            namespace: metadata.namespace,
+            annotations: metadata.annotations,
+            finalizers: metadata.finalizers,
+            ..Default::default()
+        };
+
+        store.insert(key, k8_obj.clone()).await?;
+
+        Ok(k8_obj)
+    }
+
+    /// update status
+    async fn update_status<S>(
+        &self,
+        value: &UpdateK8ObjStatus<S>,
+    ) -> Result<K8Obj<S>, Self::MetadataClientError>
+    where
+        S: Spec,
+    {
+        let store = self.get_store::<S>().await;
+
+        let key = value.metadata.name.clone();
+        debug!(key,?value.status,"start updating status");
+
+        let k8_value: Option<K8Obj<S>> = store.get(&key).await?;
+        let k8_value = k8_value.ok_or(MemoryClientError::NotFound)?;
+
+        let k8_obj = k8_value.set_status(value.status.clone());
+        debug!(key,?value.status,"overwrite set");
+
+        store.insert(key, k8_obj.clone()).await?;
+
+        debug!("done");
+
+        Ok(k8_obj)
+    }
+
+    /// patch existing with spec
+    async fn patch<S, M>(
+        &self,
+        _metadata: &M,
+        _patch: &serde_json::Value,
+        _merge_type: PatchMergeType,
+    ) -> Result<K8Obj<S>, Self::MetadataClientError>
+    where
+        S: Spec,
+        M: K8Meta + Display + Send + Sync,
+    {
+        // In SC, this api is not call
+        unreachable!()
+    }
+
+    /// patch status
+    async fn patch_status<S, M>(
+        &self,
+        _metadata: &M,
+        _patch: &serde_json::Value,
+        _merge_type: PatchMergeType,
+    ) -> Result<K8Obj<S>, Self::MetadataClientError>
+    where
+        S: Spec,
+        M: K8Meta + Display + Send + Sync,
+    {
+        // In SC, this api is not call
+        unreachable!()
+    }
+
+    /// stream items since resource versions
+    fn watch_stream_since<S, N>(
+        &self,
+        _namespace: N,
+        _resource_version: Option<String>,
+    ) -> BoxStream<'_, TokenStreamResult<S, Self::MetadataClientError>>
+    where
+        S: Spec + 'static,
+        S::Status: 'static,
+        S::Header: 'static,
+        N: Into<NameSpace>,
+    {
+        let ft_stream = async move {
+            let kind: String = S::kind();
+
+            let mut lock = self.data.lock().await;
+            let store = lock.entry(kind).or_default();
+            let st2 = store.clone();
+            drop(lock);
+            st2.watch_stream()
+        };
+
+        ft_stream.flatten_stream().boxed()
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum MemoryClientError {
```

Comment:
I think it is better to migrate to anyhow

Reply:
The issue is that the trait `MetadataClientError` has several [constraints](https://github.com/infinyon/k8-api/blob/577ad7e3894ea2cf3317461ec1a43dafa841a916/src/k8-metadata-client/src/client.rs#L97C1-L103C1): , and using anyhow would require 

    type MetadataClientError: MetadataClientError
        + Send
        + Display
        + From<IoError>
        + From<DiffError>
        + From<SerdeJsonError>;